src/daft-micropartition/src/micropartition.rs

```diff
@@ -130,11 +130,16 @@ fn materialize_scan_task(
                 FileFormatConfig::Parquet(ParquetSourceConfig {
                     coerce_int96_timestamp_unit,
                     field_id_mapping,
+                    row_groups,
                 }) => {
                     let inference_options =
                         ParquetSchemaInferenceOptions::new(Some(*coerce_int96_timestamp_unit));
                     let urls = urls.collect::<Vec<_>>();
-                    let row_groups = parquet_sources_to_row_groups(scan_task.sources.as_slice());
+                    let row_groups = if let Some(row_groups) = row_groups {
+                        Some(row_groups.clone())
+                    } else {
+                        parquet_sources_to_row_groups(scan_task.sources.as_slice())
+                    };
```

Comment:
Does this logic also need to be fixed? I think it should similarly be using the rowgroups in the scantask rather than the rowgroups on the ParquetSourceConfig.

I think it's not triggering right now because your Micropartitions aren't lazy -- they're being eagerly materialized in `from_scan_task -> read_parquet_into_micropartition`, so `materialize_scan_task` isn't ever being called...

Reply:
is there a good way to test the `materialize_scan_task` path?

Reply:
Usually I do it by:

1. Adding a filter predicate will trigger the eager materialization path
2. Omitting a filter predicate *should* trigger the lazy `materialize_scan_task` path

I'm actually not sure why in your case it doesn't seem to trigger this code-path... Since you should be in the (2) camp.

Reply:
it looks like it eagerly materializes if the parquet file does not contain statistic.

Reply:
I see... I have wondered if it might be a better idea to just always go the lazy route and simplify the code here. 

Maybe @samster25 might know why we don't do that?

Definitely very bug prone right now, I've had to tiptoe around this code a lot.

Reply:
yeah the parquet logic in general has a lot of extra logic branches not in the other file types that make it a bit hard to debug. Definitely a good candidate for some cleanup/refactoring.